include/logging/log_core.h

```diff
@@ -263,10 +263,35 @@ extern "C" {
 /*****************************************************************************/
 /****************** Definitions used by minimal logging *********************/
 /*****************************************************************************/
+#ifdef CONFIG_LOG_MODE_MINIMAL
+
 void z_log_minimal_hexdump_print(int level, const void *data, size_t size);
 void z_log_minimal_vprintk(const char *fmt, va_list ap);
 void z_log_minimal_printk(const char *fmt, ...);
 
+#else
+
+static inline void z_log_minimal_hexdump_print(int level, const void *data, size_t size)
```

Comment:
This may be a dumb question, but why not have a macro version that converts to either z_log_minimal_xxxx() or an empty macro if CONFIG_LOG_MODE_MINIMAL is defined or not? Even in the -O0 (no optimization), the macro method should not add any overhead--I am less sure off hand about how the static inline method will behave in that circumstance, but my gut tells me that the compiler will do exactly as told and leave it in.